### PR TITLE
wibox: Add more constructor arguments

### LINKED ISSUE #916: Wibar ignores some arguments for wibox construction
for example, `visible` is hardcoded to `true` so you have to set it after creating a widget

and the rest of args is just skipped:
```
border_width
border_color
ontop 	
cursor 	
opacity 
x 	
y 	
drawable 	
widget 	
window 	
shape_bounding 
shape_clip     
bg 	
bgimage 
fg 	
```

mb in `wibar.new()` we can iterate through args and assign them all to a newly created wibar?

### PR DESCRIPTION
**Pull request by Elv13**
Fixes #916

### CODE REVIEW

File: tests/run.sh [hunk 1]
```diff
@@ -17,10 +17,14 @@ if [ "$CI" = true ]; then
...
-source_dir="$PWD"
+source_dir="$(dirname "$(realpath "$(type -p $0)")")"/..
 build_dir="$CMAKE_BINARY_DIR"
 if [ -z "$build_dir" ]; then
-    build_dir="$PWD/build"
+    if [ -d "$source_dir/build" ]; then
+        build_dir="$source_dir/build"
+    else
+        build_dir="$PWD"
+    fi
```
Comment (psychon) [inline tests/run.sh]: Did this commit end up in this PR on purpose? My comments from the last time that I saw this patch still apply (e.g. bashism that doesn't work in zsh; I don't see how this worked before my patch, ...)

Comment (Elv13) [inline tests/run.sh]: No, it wasn't supposed to be there. I still need this to run the tests using my setup (and the sleep 0.5), I accidentally pushed it again.

**Comment by Elv13** [comment]
Ok, this should be it

**Comment by psychon** [comment]
:+1:

**Comment by actionless** [comment]
so now i can assign `.align` property instead of calling placement function explicitly, right?

File: lib/awful/wibar.lua [hunk 2]
```diff
@@ -205,11 +205,14 @@ function awfulwibar.set_position(wb, position, screen) --luacheck: no unused arg
 end
 
 --- Attach a wibox to a screen.
--- If a wibox is attached, it will be automatically be moved when other wiboxes
--- will be attached.
+--
+-- This function has been moved to the `awful.placement` module. Calling this
+-- no longer do anything.
```
Comment (blueyed) [inline lib/awful/wibar.lua]: s/do/does/

**Comment by Elv13** [comment]
> so now i can assign .align property instead of calling placement function explicitly, right?

The `placement` API is the recommended way to place things (wibox/client/tooltip/menu/mouse/...). `awful.wibar.align` is still deprecated. Those placement function are shared across many modules. The goal of this API is to pack together all geometric placing/resizing function so they can be composed into more complex ones and share helper code to support corner cases like screen resize and workarea changes that would have been otherwise implemented in a "zillion" places.

**Pull request merged**